source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -349,6 +352,10 @@ Http::FilterDataStatus JsonTranscoderFilter::encodeData(Buffer::Instance& data,
   readToBuffer(*transcoder_->ResponseOutput(), data);
 
   if (!method_->server_streaming() && !end_stream) {
+    if (hasHttpBodyAsOutputType()) {
```

Comment:
Can we check this earlier (preferred in `decodeData` right after creating `transcoder_`) and store in a member variable?

Reply:
:+1: